sycl/source/detail/graph_impl.hpp

```diff
@@ -305,6 +326,11 @@ class graph_impl {
   std::unordered_map<std::shared_ptr<sycl::detail::event_impl>,
                      std::shared_ptr<node_impl>>
       MEventsMap;
+  /// Map for every in-order queue thats recorded a node to the graph, what
+  /// the last node added was. We can use this to create new edges on the last
+  /// node if any more nodes are added to the graph from the queue.
+  std::unordered_map<sycl::detail::queue_impl *, std::shared_ptr<node_impl>>
```

Comment:
```suggestion
  std::unordered_map<sycl::detail::queue_impl *, std::shared_ptr<node_impl>>
```
Are there any safety concerns with using a raw pointer to the `queue_impl` here? Can we guarantee that it will never be accessed after the `queue_impl` has been destroyed?

Reply:
I think it's safe because we're only using the map inside get/set functions which pass in a shared_ptr that stays alive for the lifetime of the get/set function wile the map is being accessed.

Managed to update to use weak_ptr though by using a map with `std::owner_less` rather than unordered map.